typescript/cli/src/tests/ethereum/warp/warp-rebalancer.e2e-test.ts

```diff
@@ -400,6 +406,129 @@ describe('hyperlane warp rebalancer e2e tests', async function () {
     await startRebalancerAndExpectLog('Rebalancer started successfully 🚀');
   });
 
+  it('should skip when inflight detected by explorer', async () => {
+    // Simple mock GraphQL server
+    const http = await import('http');
+    const server = http.createServer((req, res) => {
+      if (req.method === 'POST') {
+        res.writeHead(200, { 'Content-Type': 'application/json' });
+        // Return a non-empty message_view to simulate inflight
+        res.end(JSON.stringify({ data: { message_view: [{ msg_id: '1' }] } }));
+      } else {
+        res.statusCode = 404;
+        res.end();
+      }
+    });
+    await new Promise<void>((resolve) => server.listen(0, resolve));
+    const address: any = server.address();
+    const explorerUrl = `http://127.0.0.1:${address.port}`;
+
+    try {
+      // Ensure there is a potential route by creating an imbalance
+      writeYamlOrJson(REBALANCER_CONFIG_PATH, {
+        warpRouteId,
+        strategy: {
+          rebalanceStrategy: RebalancerStrategyOptions.Weighted,
+          chains: {
+            [CHAIN_NAME_2]: {
+              weighted: { weight: '25', tolerance: '0' },
+              bridge: ethers.constants.AddressZero,
+              bridgeLockTime: 1,
+            },
+            [CHAIN_NAME_3]: {
+              weighted: { weight: '75', tolerance: '0' },
+              bridge: ethers.constants.AddressZero,
+              bridgeLockTime: 1,
+            },
+          },
```

Comment:
nit: might be better to have the config defined into its own variable with the type annotation and then pass it to the write function, this way if the config changes because we add a new field or rename an already existing one the compiler will complain and it will be easier to debug why the test is failing for someone not familiar with the rebalancer code. 

```typescript
const config: RebalancerConfig = {...};


writeYamlOrJson(REBALANCER_CONFIG_PATH, config);
```